packages/server/src/adapters/next-app-dir/rethrowNextErrors.ts

```diff
@@ -1,24 +1,60 @@
-import { isNotFoundError } from 'next/dist/client/components/not-found';
-import { isRedirectError } from 'next/dist/client/components/redirect';
-import {
-  notFound as __notFound,
-  redirect as __redirect,
-} from 'next/navigation';
+import * as nextNavigation from 'next/navigation.js';
```

Comment:
without the `.js` we get this error:

![CleanShot 2024-06-01 at 14 35 03@2x](https://github.com/trpc/trpc/assets/51714798/f3509ba1-bfbd-4ac0-8298-77bf89eaa1a9)